src/components/Dashboard/Dashboard.jsx

```diff
@@ -1,163 +1,200 @@
 // import LinearProgress from "@material-ui/core/LinearProgress";
-import { useStyleDashboard } from "./Widgets/Style";
-import FiltersFlat from "./Widgets/FiltersFlat";
-import { ProfileContext } from '../../profileContext' 
-
+import { useStyleDashboard } from "./Widgets/Style"
+import FiltersFlat from "./Widgets/FiltersFlat"
+import { ProfileContext } from "../../profileContext"
+import { db } from "../../firebase"
+import CircularProgress from "@material-ui/core/CircularProgress"
 
 export default function Dashboard() {
-  const [profile, ] = useContext(ProfileContext)
-
-  const [state, setState] = React.useState({
-    mobileView: false,
-    mediumView: false,
-  });
+	const [profile] = useContext(ProfileContext)
+	const [posts, setPosts] = useState()
+	const [doubleFilter, setDubleFilter] = React.useState({})
+	const [state, setState] = React.useState({
+		mobileView: false,
+		mediumView: false
+	})
 
-  const { mobileView, mediumView } = state;
+	const { mobileView, mediumView } = state
+	const handlePosts = (filterState) => {
+		// to handle both the filters and cities queries.
+		if (filterState.city === true) {
+			const { province } = filterState
+			setDubleFilter({ ...doubleFilter, province })
+		} else if (filterState.province != undefined) {
+			const { color, category, Status, province } = filterState
+			setDubleFilter({ ...doubleFilter, color, category, Status, province })
+		} else {
+			const { color, category, Status } = filterState
+			setDubleFilter({ ...doubleFilter, color, category, Status })
+		}
+	}
+	React.useEffect(() => {
+		const res = []
+		// console.log(doubleFilter)
+		const query = db.collection("posts")
+		const categoryQuery = doubleFilter.category
+			? query.where("category ", "==", doubleFilter["category"])
+			: query
+		const colorQuery = doubleFilter.color
+			? categoryQuery.where("color", "==", doubleFilter["color"])
+			: categoryQuery
+		const provinceQuery = doubleFilter.province
+			? colorQuery.where("province", "==", doubleFilter["province"])
+			: colorQuery
+		const check = doubleFilter["Status"] === "true" ? true : false
+		const isLostQuery = doubleFilter.Status
+			? provinceQuery.where("isLost", "==", check)
+			: provinceQuery
 
-  React.useEffect(() => {
-    const setResponsiveness = () => {
-      if (window.innerWidth < 812) {
-        setState((prevState) => ({ ...prevState, mobileView: true }));
-      }
-      if (window.innerWidth < 1210) {
-        setState((prevState) => ({ ...prevState, mediumView: true }));
-      }
+		isLostQuery.get().then((snapshot) => {
+			snapshot.docs.map((doc) => res.push(doc.data()))
+			setPosts(res)
+		})
+	}, [doubleFilter])
+	React.useEffect(() => {
+		const setResponsiveness = () => {
+			if (window.innerWidth < 812) {
+				setState((prevState) => ({ ...prevState, mobileView: true }))
+			}
+			if (window.innerWidth < 1210) {
+				setState((prevState) => ({ ...prevState, mediumView: true }))
+			}
```

Comment:
Why is responsiveness being handled like this!

Reply:
I fixed this in my PR